src/utils/natives.js

```diff
@@ -0,0 +1,29 @@
+export const nativeJSONParse = window.JSON.parse;
+
+export const nativeXmlHttpOpen = XMLHttpRequest.prototype.open;
+
+// Some extensions like AdBlock override the Object.defineProperty function to prevent a redefinition of the 'ytInitialPlayerResponse' variable by YouTube.
+// But we need to define a custom descriptor to that variable to intercept its value. This behavior causes a race condition depending on the execution order with this script :(
+// This function tries to restore the native Object.defineProperty function...
+export const nativeObjectDefineProperty = (() => {
+    // Check if the Object.defineProperty function is native (original)
+    if (Object.defineProperty?.toString().indexOf("[native code]") > -1) {
+        return Object.defineProperty;
+    }
+
+    // if the Object.defineProperty function is already overidden, try to restore the native function from another window...
+    const tempFrame = createElement("iframe", { style: `display: none;` });
+    document.documentElement.append(tempFrame);
+
+    const native = tempFrame?.contentWindow?.Object?.defineProperty;
+
+    tempFrame.remove();
+
+    if (native) {
```

Comment:
In what circumstance would this ever fail? The only failing point would be creating the iframe or somehow retrieving the function. Which both don't seem very realistic to me.

Reply:
yes, my thought was that the function might be disposed after the iframe is removed. but the check doesn't help against that anyway. Therefore the if can be removed.